pkg/utils/secrets/manager/manager.go

```diff
@@ -95,6 +95,15 @@ type (
 		dataChecksum               string
 		lastRotationInitiationTime int64
 	}
+
+	// Rotation specifies certain configuration options for the rotation of secrets.
+	Rotation struct {
+		// NoCASecretAutoRotation states whether CA secrets should be skipped for automatic rotation.
+		NoCASecretAutoRotation bool
```

Comment:
I would go for a positively-named bool flag, which is always easier to understand.
And probably `false` is a sensible default.
```suggestion
		// CASecretAutoRotation states whether CA secrets are considered for automatic rotation (defaults to false).
		CASecretAutoRotation bool
```